ipn/ipnlocal/peerapi.go

```diff
@@ -1090,6 +1091,10 @@ func (h *peerAPIHandler) handlePeerPut(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, errNoTaildrop.Error(), http.StatusInternalServerError)
 		return
 	}
+	if ((distro.Get() == distro.Unraid) && (!h.ps.directFileMode)) {
```

Comment:
I don't think this should be here: if they haven't configured a shared Taildrop folder, that shouldn't turn into a mysterious internal server error.

Reply:
I'm not sure what the best behavior here is. The fundamental problem is that Unraid boots from a flash drive and runs from an in-memory image. There just isn't a good place to cache files for retrieval if a folder isn't configured, so the best option (to me) seems to be that any attempt to drop files should fail until a path is configured/available.

Reply:
I updated the error message for this... attempting to Taildrop when the folder is not configured will still result in an internal server error, but the error message should be less mysterious now.

Reply:
Mainly: I would like to not get support tickets that Unraid taildrop isn't working.

In tailscale/health is a health check package, which is output in `tailscale status` and surfaced to the support team. If you can add a health check warning that taildrop won't work because no shared volume has been set up, that will give us something to work on tickets involving Unraid.

Reply:
That makes sense.

The logical place to put the health check is right next to this log:

https://github.com/dkaser/tailscale/blob/687f3872ff4820f4d8907600063d78137012ca79/cmd/tailscaled/taildrop.go#L26

With that being the case, a question -- I can limit the warnable to Unraid, but it seems like it would be beneficial to set for the other NAS platforms too. Which would you prefer?